egs/wsj/s5/steps/durmod/durmodel_prepare_examples.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Copyright 2015 Hossein Hadian
+
+min_repeat_count=2
+max_duration=0
+left_context=4
+right_context=2
+cmd=run.pl
+
+[ -f ./path.sh ] && . ./path.sh; # source the path.
+. parse_options.sh || exit 1;
+
+if [ $# != 3 ]; then
+   echo "Usage: $0 [options] <phones-dir> <ali-dir> <duration-model-dir>"
+   echo "... where <duration-model-dir> is assumed to be a sub-directory "
+   echo " of the directory where the transition model (i.e. final.mdl) is."
+   echo "e.g.: $0 data/lang/phones exp/mono_ali exp/mono/durmod"
+   echo ""
+   echo "Main options (for others, see top of script file):"
+   echo "  --min-repeat-count <count>                     # minimum number of occurence for a duration; used to determine max-duration"
+   echo "  --max-duration <duration-in-frames>            # max duration; if not set, it will be determined automatically"   
+   echo "  --left-context <size>                          # left phone context size"   
+   echo "  --right-context <size>                         # right phone context size"   
+   exit 1;
+fi
+
+phonesdir=$1
+alidir=$2
+dir=$3
+
+srcdir=`dirname $dir`
+durmodel=$dir/durmodel.mdl
+transmodel=$srcdir/final.mdl
+
+for f in $transmodel $alidir/ali.1.gz $phonesdir/roots.int $phonesdir/extra_questions.int; do
+  [ ! -f $f ] && echo "$0: Required file not found: $f" && exit 1;
+done
+
+rm $dir/all.ali 2>/dev/null
+mkdir -p $dir/log || exit 1;
+gunzip -c $alidir/ali.*.gz >> $dir/all.ali || exit 1;
+
+if [ $max_duration == 0 ]; then
+  echo "$0: Determining max-duration..."
+  $cmd $dir/log/ali_to_phones.log \
+       ali-to-phones --write-lengths $transmodel ark:$dir/all.ali \
+                     ark,t:$dir/all.length_ali || exit 1;
```

Comment:
I would be fine with using awk for this.  Just make sure your script is
portable to e.g. BSD versions of awk, i.e. no fancy features.
Dan


On Tue, Jan 12, 2016 at 1:32 PM, Hossein Hadian <notifications@github.com>
wrote:

> In egs/wsj/s5/steps/durmod/durmodel_prepare_examples.sh
> <https://github.com/kaldi-asr/kaldi/pull/414#discussion_r49516908>:
>
> > +durmodel=$dir/durmodel.mdl
> > +transmodel=$srcdir/final.mdl
> > +
> > +for f in $transmodel $alidir/ali.1.gz $phonesdir/roots.int $phonesdir/extra_questions.int; do
> > +  [ ! -f $f ] && echo "$0: Required file not found: $f" && exit 1;
> > +done
> > +
> > +rm $dir/all.ali 2>/dev/null
> > +mkdir -p $dir/log || exit 1;
> > +gunzip -c $alidir/ali.*.gz >> $dir/all.ali || exit 1;
> > +
> > +if [ $max_duration == 0 ]; then
> > +  echo "$0: Determining max-duration..."
> > +  $cmd $dir/log/ali_to_phones.log \
> > +       ali-to-phones --write-lengths $transmodel ark:$dir/all.ali \
> > +                     ark,t:$dir/all.length_ali || exit 1;
>
> No it is not used anywhere else. I will pipe it.
> Actually the script finds the max duration d, for which all durations d-1,
> d-2, ... have occurred at least $min_repeat_count times. I thought a python
> script would be cleaner. Also in the future we might want to determine
> max-duration using some other method, for eg. by finding the variance and
> mean of durations. Should I write it in awk ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/414/files#r49516908>.
>